crawlerr/services/gpu_manager.py

```
@@ -0,0 +1,338 @@
+    def is_gpu_available(self) -> bool:
+        """Check if GPU acceleration is available and healthy."""
+        return self.gpu_available and self._gpu_healthy
+
+    def should_use_gpu(self) -> bool:
+        """Determine if GPU should be used for current operation."""
+        if not self.is_gpu_available():
+            return False
+
+        # Additional logic can be added here for dynamic GPU usage decisions
+        # For now, use GPU if available and healthy
+        return True
```

Ruff (0.12.2)

Return the negated condition directly

(SIM103)